mcs/class/System/System.Net/HttpListenerRequest.cs

```diff
@@ -403,7 +403,7 @@ internal bool FlushInput ()
 		}
 
 		public bool IsLocal {
-			get { return IPAddress.IsLoopback (RemoteEndPoint.Address); }
+			get { return LocalEndPoint.Address.Equals (RemoteEndPoint.Address); }
```

Comment:
OK, if we're still in for bug-to-bug compatibility, then I guess it's fine. I still don't think this behavior is correct, though :)